### PR TITLE
fix: relax type expectations for hparam values [DET-4742 DET-4744]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

hyperparameter values can be any valid yaml types https://docs.determined.ai/latest/reference/experiment-config.html#hyperparameters

## Test Plan

added tests.

revert the fix commit and run the test to see the failure. 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by hamidzr** [comment]
@katport I tried the 
[adaptive.txt](https://github.com/determined-ai/determined/files/5784734/adaptive.txt) config. it loaded fine and I added it as a test case. Could you also please try it on a local build or any other experiment configs you suspect might fail related to this issue?

**Review by hkang1 — APPROVED** [review]
LGTM! noting one small change of stamping out the 5 different types we encounter for hparam types (adding categorical).

File: webui/react/src/types.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ interface ExperimentHyperParam {
   maxval?: number;
   minval?: number;
   type: string;
-  val?: number | string;
+  val?: unknown;
```
Comment (katport) [inline webui/react/src/types.ts:187]: Does this include Nonetype? i.e. 

hp_var: None

Comment (katport) [reply webui/react/src/types.ts:187]: Same with 
Categorical:
 types: ['a', None]

Comment (hamidzr) [reply webui/react/src/types.ts:187]: I think they'd come through as strings since `None` is not a special value in YAML. We seem to have some samples

**Review by katport — APPROVED** [review]

File: webui/react/src/types.ts (hunk 1 above)
Comment (hamidzr) [reply webui/react/src/types.ts:187]: https://github.com/determined-ai/determined/pull/1763/files#diff-98b216143056cb9a93b0e5179a1b513607c33cdcd4052fb2b2aec880c370bff3R1125

ahh we don't have `types: ['a', None]` I think

Comment (katport) [reply webui/react/src/types.ts:187]: let me clarify as I was shorthanding it a bit: 

HP_var1:
   type: categorical
   vars: ['a', None]

Comment (hamidzr) [reply webui/react/src/types.ts:187]: I'm confused about the None usage here. 

I'm assuming https://github.com/determined-ai/determined/pull/1763#discussion_r554128598 
translates to json like so: 

```json
  {
    "myParam": {
      "type": "categorical",
      "vals": ["a", "None"]
    }
  },
￼

Comment (hamidzr) [reply webui/react/src/types.ts:187]: I added a test case with None https://github.com/determined-ai/determined/pull/1763/commits/4bda23bec2578b2897bb3d8145b47fe3fa412419


I just checked to make sure `vars: ['a', None]` the quotations, or lack there of,  don't play a role here.

Screen Capture_select-area_20210108112556

**Comment by hamidzr** [comment]
I'm setting up a test stack here https://github.com/determined-ai/determined/pull/1771

**Pull request merged**